test_keyboard_interrupt_does_not_resume_test times out on weak hardware

### Things to check first

- [x] I have searched the existing issues and didn't find my bug already reported there

- [x] I have checked that my bug is still present in the latest release


### AnyIO version

4.14.2

### Python version

3.14.6

### What happened?

Running the test suite on a Power Mac G5 from 2005:

```
_________________ test_keyboard_interrupt_does_not_resume_test _________________
[gw1] linux -- Python 3.14.6 /nix/store/sqkvf269mvjm2md4cb9lq7835yxdz6am-python3-3.14.6/bin/python3.14
/nix/store/sqkvf269mvjm2md4cb9lq7835yxdz6am-python3-3.14.6/lib/python3.14/subprocess.py:1279: in wait
    return self._wait(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/sqkvf269mvjm2md4cb9lq7835yxdz6am-python3-3.14.6/lib/python3.14/subprocess.py:2076: in _wait
    raise TimeoutExpired(self.args, timeout)
E   subprocess.TimeoutExpired: Command '('/nix/store/sqkvf269mvjm2md4cb9lq7835yxdz6am-python3-3.14.6/bin/python3.14', '-mpytest', '--basetemp=/build/pytest-of-nixbld/pytest-0/popen-gw1/test_keyboard_interrupt_does_not_resume_test0/runpytest-0', '-v', '-p', 'anyio', '-p', 'no:asyncio', '-p', 'no:trio')' timed out after 5 seconds

During handling of the above exception, another exception occurred:
/build/source/tests/test_pytest_plugin.py:507: in test_keyboard_interrupt_does_not_resume_test
    result = testdir.runpytest_subprocess(*pytest_args, timeout=5)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/nix/store/0abnlfhrjf26cwd0vijw6cm3n0dzjr3w-python3.14-pytest-9.0.3/lib/python3.14/site-packages/_pytest/legacypath.py:236: in runpytest_subprocess
    return self._pytester.runpytest_subprocess(*args, timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   _pytest.pytester.Pytester.TimeoutExpired: 5 second timeout expired running: ('/nix/store/sqkvf269mvjm2md4cb9lq7835yxdz6am-python3-3.14.6/bin/python3.14', '-mpytest', '--basetemp=/build/pytest-of-nixbld/pytest-0/popen-gw1/test_keyboard_interrupt_does_not_resume_test0/runpytest-0', '-v', '-p', 'anyio', '-p', 'no:asyncio', '-p', 'no:trio')
----------------------------- Captured stdout call -----------------------------
running: /nix/store/sqkvf269mvjm2md4cb9lq7835yxdz6am-python3-3.14.6/bin/python3.14 -mpytest --basetemp=/build/pytest-of-nixbld/pytest-0/popen-gw1/test_keyboard_interrupt_does_not_resume_test0/runpytest-0 -v -p anyio -p no:asyncio -p no:trio
     in: /build/pytest-of-nixbld/pytest-0/popen-gw1/test_keyboard_interrupt_does_not_resume_test0
```

Raising the timeout value [here](https://github.com/agronholm/anyio/blob/caca0e076d4052fca751f1a6b5e248bb4901f6cd/tests/test_pytest_plugin.py#L507) from 5 seconds to smth like 15 seconds allows the test to pass on this machine.

Raising the timeout doesn't *seem* like it would affect the logic of the test? Maybe it could just be dropped altogether?

### How can we reproduce the bug?

Run the `test_keyboard_interrupt_does_not_resume_test` test on old / comparatively weak hardware

--- a/tests/test_pytest_plugin.py
+++ b/tests/test_pytest_plugin.py
@@ -499,7 +499,7 @@ def test_keyboard_interrupt_does_not_resume_test(testdir: Pytester) -> None:
         async def test_keyboard_interrupt(myfixture):
             loop = asyncio.get_running_loop()
             loop.call_soon(signal.raise_signal, signal.SIGINT)
-            await anyio.sleep(3)
+            await anyio.sleep(5)
             print("RESUMED_AFTER_INTERRUPT")
         """
     )

--- a/tests/test_pytest_plugin.py
+++ b/tests/test_pytest_plugin.py
@@ -499,12 +499,12 @@ def test_keyboard_interrupt_does_not_resume_test(testdir: Pytester) -> None:
         async def test_keyboard_interrupt(myfixture):
             loop = asyncio.get_running_loop()
             loop.call_soon(signal.raise_signal, signal.SIGINT)
-            await anyio.sleep(3600)
+            await anyio.sleep(3)
             print("RESUMED_AFTER_INTERRUPT")
         """
     )
 
-    result = testdir.runpytest_subprocess(*pytest_args, timeout=5)
+    result = testdir.runpytest_subprocess(*pytest_args)
     assert result.ret == 2
     result.stdout.no_fnmatch_line("*RESUMED_AFTER_INTERRUPT*")
     result.stdout.fnmatch_lines(["*KeyboardInterrupt*"])

